lib/lint.js

```diff
@@ -0,0 +1,115 @@
+const fs = require('fs-extra');
+const exec = require('child-process-promise').exec;
+const walkSync = require('walk-sync');
+
+const STANDARD_APP_FILES = ['index.js', 'ember-cli-build.js', 'testem.js'];
+
+const STANDARD_APP_PATHS = [
+  'addon',
+  'addon-test-support',
+  'app',
+  'blueprints',
+  'config',
+  'lib',
+  'server',
+  'test-support',
+  'tests',
+];
+const STANDARD_STYLE_PATHS = ['styles', 'addon/styles', 'app/styles', 'tests/dummy/app/styles'];
+
+function lintFileNames(root, paths = []) {
+  if (paths.length === 0) {
+    for (let path of STANDARD_APP_PATHS) {
+      if (fs.existsSync(`${root}/${path}`)) {
+        paths.push(...walkSync(`${root}/${path}`));
+      }
+    }
+  }
+
+  let matches = paths.filter(path => !path.match(/^(@?[a-z-]+\/)*(_?[a-z-.]+)?$/));
+
+  if (matches.length > 0) {
+    return [
+      false,
+      `Filename lint failed, found malformed file names. All files must be kebab case: \n${matches.join(
+        '\n'
+      )}`,
+    ];
+  } else {
+    return [true, 'Filename lint passed! 🎉'];
+  }
+}
+
+function lintJavascript(root, paths = []) {
+  if (paths.length === 0) {
+    paths = STANDARD_APP_PATHS.concat(STANDARD_APP_FILES);
+  }
+
+  let eslintPath = require.resolve('eslint/bin/eslint');
+
+  return exec(`${eslintPath} ${paths.join(' ')}`, { cwd: root })
+    .then(() => {
+      return [true, 'Javascript lint passed! 🎉'];
+    })
+    .catch(err => {
+      return [false, `Javascript lint failed:\n${err.stdout}`];
+    });
+}
+
+function lintSass(root, paths = []) {
+  if (paths.length === 0) {
+    paths = STANDARD_STYLE_PATHS;
+  }
+
+  let sassLintPath = require.resolve('sass-lint/bin/sass-lint');
+
+  return exec(`${sassLintPath} -vq`, { cwd: root })
```

Comment:
all calls to `exec` look the same, I wonder if we can get away with creating a `runLint` function to DRY the code up a little?

```javascript
function runLint(name, options) {
  let { command, root } = options;

  return exec(command, { cwd: root })
    .then(() => [true, `${name} lint passed! 🎉`])
    .catch(e => [false, `${name} lint failed:\n${e.stdout}`])
}